From david.engraf@netcom.eu Thu Mar 20 02:54:19 2008
From: David Engraf <david.engraf@netcom.eu>
Date: Thu, 20 Mar 2008 10:53:52 +0100
Subject: USB: cdc-acm tell tty layer not to split things up.
To: Greg KH <greg@kroah.com>
Cc: linux-usb@vger.kernel.org
Message-ID: <47E23430.9010401@netcom.eu>


It ensures that the tty level do not split
the send buffer into 2KB blocks.


Signed-off-by: David Engraf <david.engraf@netcom.eu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/cdc-acm.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -489,6 +489,7 @@ static int acm_tty_open(struct tty_struc
 	else
 		rv = 0;
 
+	set_bit(TTY_NO_WRITE_SPLIT, &tty->flags);
 	tty->driver_data = acm;
 	acm->tty = tty;
 
